apps/src/lib/tools/jsdebugger/DebugConsole.jsx

```diff
@@ -201,10 +214,20 @@ export default connect(
       this.props.appendLog(output);
     }
 
-    componentDidUpdate() {
-      this._debugOutput.scrollTop = this._debugOutput.scrollHeight;
+    componentDidUpdate(prevProps) {
+      const prevOutputSize = prevProps.logOutput.size;
+      if (
+        typeof prevOutputSize === 'number' &&
+        prevOutputSize !== this.props.logOutput.size
+      ) {
+        this.jumpToBottom();
+      }
     }
 
+    jumpToBottom = () => {
+      this._debugOutput.scrollTop = this._debugOutput.scrollHeight;
+    };
```

Comment:
note to reviewers: this is the functional change -- the rest of the changes in this file are clean-up